docs/scos/dev/the-docker-sdk/202204.0/the-docker-sdk.md

```diff
@@ -15,12 +15,219 @@ related:
+ - Executes the `docker-compose up -d` command.
+ - Runs the demo data installation process for each region.
+
+The `docker/sdk export images {tag}` command is only for the baked mode. It does the following:
+ - Builds and tags docker application images.
+ - Builds assets builder docker image based on assetsBuilder and Cli image). Runs `vendor/bin-install -s build-static build-static-{mode}`.
+ - Builds and tags frontend images.
+ - Prints information about the built images.
+
+## Development environment
+The following schema illustrates the development environment within the Docker SDK:
+![development-environment](https://spryker.s3.eu-central-1.amazonaws.com/docs/scos/dev/the-docker-sdk/the-docker-sdk/development-environment.png)
+
+In the development environment, Gateway (Nginx) does the following:
+
+* Accepts all public HTTP requests.
+* Handles SSL connections.
+* Serves all non-application services.
+* Proxies application calls to Frontend (Nginx).
+
+Fontend (Nginx), translates HTTP to FastCGI proxy for Glue, Yves, Zed, and serves assets.
+
+## Docker/sdk debug mode
+
+The following schema illustrates the Docker/sdk debug mode:
+
+![docker-sdk-debug-mode](https://spryker.s3.eu-central-1.amazonaws.com/docs/scos/dev/the-docker-sdk/the-docker-sdk/docker-sdk-debug-mode.png)
+
+## Docker/sdk in SCCOS
+The following schema illustrates how Docker/sdk is used in the Spryker Cloud Commerce OS (SCCOS):
+![docker-sdk-in-sccos](https://s3.console.aws.amazon.com/s3/object/spryker?region=eu-central-1&prefix=docs/scos/dev/the-docker-sdk/the-docker-sdk/docker-sdk-in-sccos.png)
+
+## Docker images
+The following schemas illustrate the Docker images:
+![docker-image-nesting](https://spryker.s3.eu-central-1.amazonaws.com/docs/scos/dev/the-docker-sdk/the-docker-sdk/Docker-image-nesting.png)
+
+Docker images and containers:
+![docker-images-and-containers](https://spryker.s3.eu-central-1.amazonaws.com/docs/scos/dev/the-docker-sdk/the-docker-sdk/Docker-files-inheritance.png)
+
+## Extending Docker/sdk
+To extend Docker/sdk, you can do the following:
+
+- To use custom PHP/nginx image and add extensions or libraries, you can change the application image:
+   
+   ```
+   environment: docker.dev
+   image:
+      tag: spryker/php 7.3
+      tag: registry.gitlab.com/my-company/spryker:1.0-php8debsap
+   ```
+  This approach supports SCCOS, but has limited customization possibilities.
+
+- To make major customizations and keep control of the code, you can fork the [docker/sdk repository](https://github.com/spryker/docker-sdk).
+  This approach offers all the customization possibilities but does not support SCCOS.
+
+- To introduce mocks for development and for CI/CD testing, you can use the [Docker-compose extension](https://docs.docker.com/compose/extends/):
+  ```
+  docker:
+   compose:
+      yamls:
+         - {custom_compose_file_name}
+
+  ```
+
+  This approach works great for dev/CI/CD, but requires SCCOS customizations.
```

Comment:
This approach works great for dev/CI/CD only, it won't take any effect on your Cloud setup.